### DISCUSSION TITLE
Oathkeeper regex rules are case sensitive - a lot web servers / apps are not

### DISCUSSION THREAD

**Discussion by viters** [ory/oathkeeper#940] [Q&A] [answered]
I think it is worth pointing out, that 
![image](https://user-images.githubusercontent.com/13100466/158597849-64204526-566e-46f3-aca1-9ed3371c7e34.png)

is rather:

`http://mydomain.com/<(?!protected).*>` does not match exactly `http://mydomain.com/protected`, but will match:
- http://mydomain.com/PROTECTED
- http://mydomain.com/Protected
- http://mydomain.com/ProTecteD
etc.

Does it create a security trap for someone configuring the rules?

By the way, I know that `oathkeeper` uses `dlclark/regexp2` underneath. It states in description that it is perl compatible. I am not sure if `(?!protected).*` is perl-compatible, because negative lookahead should only affect letter `p` with that regex:
![image](https://user-images.githubusercontent.com/13100466/158598928-94be3dc1-99a5-49f7-8411-b78e6da278bb.png)
`http://mydomain.com/<(?!protected).*>` would not match any URL with path starting with lower-case `p`.

**Comment by viters** [comment, marked as answer]
@aeneasr Probably would be useful to have your opinion here

**Reply by aeneasr** [reply]
Nice find, thank you @viters! I think we need to improve the docs here. Any contributions are highly appreciated :))

**Reply by vinckr** [reply]
Hello @viters,
since this discussion is stale for a while, would you be willing to [open an issue](https://github.com/ory/docs/issues/new/choose) for this in ory/docs? 
That would help us keep track of it 🙏 
I will mark this as answered, but feel free to come back to the discussion any time.

**Reply by viters** [reply]
@vinckr 
Yes, I will.
